rsopt/configuration/setup.py

```diff
@@ -208,6 +208,9 @@ def get_run_command(self, is_parallel):
 
         return run_command
 
+    def get_sym_link_targets(self) -> set:
```

Comment:
Is this method supposed to be more of an abstract base class? i.e. this should `raise NotImplementedError` and needs to be defined in a child class?

Reply:
I think this needs to always be defined and return an empty set. Otherwise we would need to implement a check to make sure it is defined.

Reply:
That's the point though -- Is a "bare" `Setup` class every supposed to be created anywhere? If it isn't, an error should be thrown as someone would be misusing the class

Reply:
`Setup` should never be instantiated anywhere, but the default behavior for `get_sym_link_targets` should be to return an empty set, as the method current does for `Setup`.
Should a default like this be explicitly written for every subclass instead of just in `Setup`?

Reply:
That's my question, basically. The classes that you created based on `Setup` both redefine this method, so I am just making sure that an arbitrary class that derives from `Setup` should return an empty set by default. If the class _should_ redefine `get_sym_link_targets`, otherwise it isn't valid, then this should instead raise an error.

I'm approving the request as only you know _that_ answer